e2e/testcases/kcc_test.go

```diff
@@ -129,3 +135,93 @@ func TestKCCResourcesOnCSR(t *testing.T) {
 		nt.T.Fatal(err)
 	}
 }
+
+func TestResourceGroupKCC(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.ACMController, ntopts.KCCTest)
+
+	ns := v1.Namespace{}
+	ns.Name = "resourcegroup-e2e"
+	nt.T.Cleanup(func() {
+		// all test resources are created in this namespace
+		if err := nt.KubeClient.Delete(ns.DeepCopy()); err != nil {
+			nt.T.Error(err)
+		}
+	})
+	if err := nt.KubeClient.Create(ns.DeepCopy()); err != nil {
+		nt.T.Fatal(err)
+	}
+	rgNN := types.NamespacedName{
+		Name:      "group-kcc",
+		Namespace: ns.Name,
+	}
+	resourceID := rgNN.Name
+	rg := testresourcegroup.NewKptResourceGroup(rgNN, resourceID)
+	if err := nt.KubeClient.Create(rg.DeepCopy()); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Create a KCC object and verify the ResourceGroup status")
+	kccObj := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "serviceusage.cnrm.cloud.google.com/v1beta1",
+			"kind":       "Service",
+			"metadata": map[string]interface{}{
+				"name":      "pubsub.googleapis.com",
+				"namespace": ns.Name,
+				"annotations": map[string]interface{}{
+					"config.k8s.io/owning-inventory": resourceID,
+				},
+			},
+		},
+	}
+	resource := v1alpha1.ObjMetadata{
+		GroupKind: v1alpha1.GroupKind{
+			Group: kccObj.GroupVersionKind().Group,
+			Kind:  kccObj.GroupVersionKind().Kind,
+		},
+		Namespace: kccObj.GetNamespace(),
+		Name:      kccObj.GetName(),
+	}
+	nt.T.Cleanup(func() {
+		if err := nt.KubeClient.Delete(kccObj.DeepCopy()); err != nil {
+			nt.T.Error(err)
+		}
+	})
+	if err := nt.KubeClient.Create(kccObj.DeepCopy()); err != nil {
+		nt.T.Fatal(err)
+	}
+	nt.T.Log("Add the KCC object to the ResourceGroup")
+	err := testresourcegroup.PatchResourceGroup(nt.KubeClient, rgNN, func(rg *v1alpha1.ResourceGroup) {
+		rg.Spec.Resources = []v1alpha1.ObjMetadata{resource}
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	nt.T.Log("Verifying the resourcegroup status: It can surface the kcc resource status.conditions")
+	expectedStatus := testresourcegroup.InitialResourceStatus()
+	expectedStatus.ObservedGeneration = 2
+	expectedStatus.ResourceStatuses = []v1alpha1.ResourceStatus{
+		{
+			ObjMetadata: resource,
+			Conditions: []v1alpha1.Condition{
+				{
+					Type:    "Ready",
+					Status:  "False",
+					Reason:  "UpdateFailed",
+					Message: "Update call failed",
+				},
+			},
+			Status: v1alpha1.InProgress,
+		},
+	}
+	err = nt.Watcher.WatchObject(kinds.ResourceGroup(), rgNN.Name, rgNN.Namespace, []testpredicates.Predicate{
+		testpredicates.ResourceGroupHasStatus(expectedStatus),
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	nt.T.Log("Delete the ResourceGroup")
+	if err := nt.KubeClient.Delete(rg.DeepCopy()); err != nil {
```

Comment:
should delete be in Cleanup or do we only want to delete on success?

Reply:
Not sure what the intent is, the upstream test deletes the ResourceGroup at the end of the test. Cleanup will manage deletion regardless by deleting the namespace.

Reply:
can you make these tests consistent? either always delete at the end or delete in cleanup.

Reply:
Removed the delete from the test(s) which aren't doing any validation after deleting the ResourceGroup.

FWIW I'm trying to stay as close as possible to the spirit of the tests in our upstream repo, so that with this initial commit we can easily map the test coverage to upstream.

Reply:
Engineering is full of conflicting motivations, opinions, and priorities.

If you think something I asked for is a waste of time feel free to push back or ask me to do it in a follow up. Just because I asked for it doesn't necessarily mean I feel strongly about it. 

I'm trying to keep project quality high but if it annoys everyone in the process then it isn't worth the cost.